### PR TITLE
feat(secrets): add Doppler adapter

### PR DESCRIPTION
**Pull request by mrbongiolo**
Doppler organizes secrets in "projects" (like `my-awesome-project`) and "configs" (like `prod`, `stg`, etc.), so the pattern `project/config` is required when defining the `--account` option.

Doppler does not have a concept of folders, so using `--from` option or `FOLDER/SECRET` pattern is not supported and will raise an error.

Initially, I was thinking of using the `--from` option to define the Doppler "config", but this would require a much larger change in how secrets are "folderized", so not allowing to use that option and parsing `--account=project/config` accordingly seemed like a better approach for now.

Site documentation PR: https://github.com/basecamp/kamal-site/pull/134

### CODE REVIEW

File: lib/kamal/secrets/adapters/doppler.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      $?.success?
+    end
+
+    def fetch_secrets(secrets, account:, session:)
+      project, config = account.split("/")
+
+      raise RuntimeError, "Missing project or config from --acount=project/config option" unless project && config
+      raise RuntimeError, "Using --from option or FOLDER/SECRET is not supported by Doppler" if secrets.any?(/\//)
+
+      items = `doppler secrets get #{secrets.map(&:shellescape).join(" ")} --json -p #{project} -c #{config}`
```
Comment (djmb) [inline lib/kamal/secrets/adapters/doppler.rb:21]: We should shell escape `project` and `config` here as well as we are injecting user input. Admittedly probably a friendly user, but best to be on the safe side.

**Comment by djmb** [comment]
Oh, could we also implement the `check_dependencies!` method that was recently added. It should give a nice error message if doppler is not installed.

**Comment by mrbongiolo** [comment]
Thanks for the review @djmb!! I'll fix the points that you brought up and update the PR later today.

**Comment by mrbongiolo** [comment]
@djmb PR updated. Also added support for `ENV["DOPPLER_TOKEN"]`, based on https://github.com/basecamp/kamal/pull/1183

**Comment by djmb** [comment]
Hey @mrbongiolo - this has been released now in Kamal 2.4.0. Could you update the docs PR and then I'll merge that?

**Comment by mrbongiolo** [comment]
Nice @djmb !! The doc PR is updated, it just requires an approval for merging. Or is there something missing?

**Comment by djmb** [comment]
No all good then, there were some merge conflicts but I've resolved them and merged the PR 👍

**Pull request merged**